src/components/Registries/RegistriesCompanies/CompanyCRUD/index.js

```diff
@@ -1,50 +1,62 @@
 import React from 'react';
-import { useHistory } from 'react-router-dom';
+import { useHistory, useParams } from 'react-router-dom';
 import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
 import { registriesSelectors } from '../../../../redux/selectors';
 import { newCompanyDataObject } from '../../../../utils/defaultData';
-import { buildRegistryForm } from '../../../../utils/registryFormBuilder';
+import { BuildRegistryForm } from '../../../../utils/registryFormBuilder';
 import { registriesActions } from '../../../../redux/actions';
 
-function CompanyCRUD() {
+function CompanyCRUD({ action }) {
```

Comment:
Instead of taking `action` as prop, let's split this into two components - one for creating, one for editing.